tests/sub_group_mask/sub_group_mask_reference.cpp

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Extension Conformance Test
+//
+//  Provides tests to check sub_group_mask operator[] const and reference
+//
+*******************************************************************************/
+
+#include "sub_group_mask_common.h"
+
+#define TEST_NAME sub_group_mask_reference
+
+namespace TEST_NAMESPACE {
+
+using namespace sycl_cts;
+#ifdef SYCL_EXT_ONEAPI_SUB_GROUP_MASK
+
+struct check_result_reference {
+  bool operator()(sycl::ext::oneapi::sub_group_mask &sub_group_mask,
+                  const sycl::sub_group &) {
+    for (size_t N = 0; N < sub_group_mask.size(); N++) {
+      auto ref = sub_group_mask[sycl::id(N)];
+      if (ref != (N % 2 == 0)) return false;
+      switch (N % 5) {
+        case 0:
+          ref = (N % 2 != 0);
+          if (sub_group_mask[sycl::id(N)] != (N % 2 != 0)) return false;
```

Comment:
What exactly this check is doing? It doesn't seem to be different from the check done before the switch.

Reply:
We verify that we changed value in `sub_group_mask` by using `ref_to_bit`